code/game/machinery/firealarm.dm

```diff
@@ -190,6 +190,7 @@
 						if(buildstage == 1)
 							if(stat & BROKEN)
 								to_chat(user, "<span class='notice'>You remove the destroyed circuit.</span>")
+								stat &= ~BROKEN

```

Comment:
For some reason I assumed that you could fix structure damage with a welding tool, but that isn't actually the case. I'll add that. As is, it fixes the issue of the alarm staying broken when a new board is inserted, but it'll break again as soon as you tap it since it can't be repaired.